### PR TITLE
State params

### PR DESCRIPTION
**Pull request by RCoder01**
Allows using State.with_params(param=value) to create a new subclass of the state with overridden parameters. Allows each transition map to use their own versions of the states--most useful for states shared among many maps and for quick prototyping.

Warns and fails typecheck.py if an unexpected parameter is passed.

Addresses #14

### CODE REVIEW

File: mrobosub_planning/src/umrsm.py [hunk 1]
```diff
@@ -56,6 +70,23 @@ def handle(self) -> Optional[NamedTuple]:
...
 
+    @classmethod
+    def with_params(cls, **kwargs: Any) -> Type['State']:
+        num_unexpected = 0
+        for k in kwargs:
+            if not hasattr(cls, k):
+                warnings.warn(f"overriding parameter {k}, which is not defined on {cls}", stacklevel=2)
+                num_unexpected += 1
+        overrides: dict = kwargs.get('_param_overrides', {}).copy()
+        overrides.update(kwargs)
```
Comment (alexanderbowler) [inline mrobosub_planning/src/umrsm.py:81]: Could this potentially lead to nesting within the _param_overrides of just more _param_overrides

Comment (RCoder01) [reply mrobosub_planning/src/umrsm.py:81]: Idts, but you made me realize what I had was wrong

Comment (alexanderbowler) [reply mrobosub_planning/src/umrsm.py:81]: is the _param_overrides just for the printing of the rendered representation if so this all seems good to me

Comment (RCoder01) [reply mrobosub_planning/src/umrsm.py:81]: Yeah it's just for the better prints

**Review by alexanderbowler — APPROVED** [review]
Looks good to me

**Pull request merged**